src/components/DetailsList/DetailsList.Props.ts

```diff
@@ -131,6 +131,9 @@ export interface IDetailsListProps extends React.Props<DetailsList> {
 
   /** Boolean value to indicate if the role application should be applied on details list. Set to false by default */
   shouldApplyApplicationRole?: boolean;
+
+  /** Optional value if you want to define the grid role on the list */
+  ariaRoleForGrid?: string;
```

Comment:
The docs here is unclear. I think we should either just leave it abstract, such as 

rootRole = controls the role attribute of the root element.
innerRole = controls the role attribute of the inner div element.

Or, we really simplify this into well document flags that explain why anyone should care. My question is, why should i care to apply an application role (shouldApplyApplicationRole)? Why is the new prop called "ariaRoleForGrid"? 

Why do we need 2 roles? (application and this new optional one) Is that even valid? If we only need 1 role, why not just have a "role" property with some sane default? Is it really necessary to even override this and specify different roles in different situations? If it's not, can we just remove all of this?

Simple is better here. This is really confusing to anyone that does not deeply understand screen reader requirements.

@cschlechty for visibility.